### PR TITLE
[networks] Serialize HTTP data as an opaque bytes field

### PR DESCRIPTION
**Pull request by p-lambert**
### What does this PR do?

Change the way we serialize HTTP data so downstream services no longer have to decode it.
Requires https://github.com/DataDog/agent-payload/pull/94

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Describe your test plan

Write there any instructions and details that should be tested during the QA.

### CODE REVIEW

**Review by ISauve — APPROVED** [review]
LGTM (pending agent-payload release ofc)

File: go.sum
Comment (brycekahle) [inline go.sum:113]: Can you try running `go mod tidy`? That should get rid of the older versions.

Comment (p-lambert) [reply go.sum:113]: Sure! will do once I merge the `agent-payload` PR and update this one

File: pkg/network/encoding/format.go
Comment (leeavital) [inline pkg/network/encoding/format.go:63]: Does this show up in an allocations profile? If so, I suggest adding a pool similar to https://github.com/DataDog/datadog-agent/pull/7789/files#diff-df016a1ba6d5c366a13582285fa42009946bf515a2f728891d64cc7cde196b80L36

Comment (p-lambert) [reply pkg/network/encoding/format.go:63]: It does look like `FormatHTTPStats` is allocating as much as `FormatConnection`.
There are a few things we could pool:
* The `map[string]*model.HTTPStats` maps;
* The `model.HTTPStats` structs;
* The `[]*model.HTTPStats_Data` slices (and by extension the `model.HTTPStats_Data` objects themselves`);

Two things to note:
1) The bad thing about caching maps is that we have to delete all entries (eg.: https://github.com/golang/go/issues/20138) though it seems that this idiom has been optimized since Golang 1.11: https://golang.org/doc/go1.11#performance-compiler 
2) I'm not entirely sure `sync.Pool` works as expected for this type of workload (given that this code is executed only once every 30s). In any case it is something I can follow up on and we can create or own pool if need

Comment (p-lambert) [reply pkg/network/encoding/format.go:63]: @leeavital I'm going to merge this as it is and address the object pool in a separate PR (already working on it).
I did some investigation and added some debugging telemetry and it turns out that `sync.Pool` is not helping at at all under memory pressure since objects are getting during GC runs (this is by design, as mentioned here https://groups.google.com/g/golang-dev/c/kJ_R6vYVYHU/m/LjoGriFTYxMJ)
I'm going to add an alternative pool implementation that will hopefully be a better fit for long lived objects

**Pull request merged**
